bin/shutter

```diff
@@ -7831,8 +7831,9 @@ if ($app->is_running) {
 
 			my $filename_template = quotemeta $filename_value;
 
-			#replace %NNN by a regex to search for digits
-			$filename_template =~ s/\\\%N+/(\d+)/g;
+			#replace %NNN by a \d+ regex to search for digits
+			#also take into account conflicted filenames Ex.: "_014(002)"
+			$filename_template =~ s/\\\%N+/(\\d+)(\\(\\d+\\))?/g;
```

Comment:
So many backslashes because we want the left part to be replaced literally by:
"(\d+)(\(\d+\))?"
That will be used itself as is in the following regexp.
So, we have to escape the backslashes themselves so that they are not "processed" as regexp special char for this first replacement.
It has been a long time the last time I did perl before, so there might be an easier way.
But, I did a few tests, and without that level of escaping the slashes, the behavior would not be as expected.

Also, in the end, with have the "outer" parenthesis for the regexp "grouping", and the "inside" parenthesis that are for the literal ones of the string "(018)".

Reply:
> It has been a long time the last time I did perl before

Yep, me too... I'm just confused why (\d+) on the left side of this diff works.